handlers/metrics.go

```diff
@@ -8,18 +8,52 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/gorilla/mux"
 	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/models"
 	"github.com/kiali/kiali/prometheus"
 	"github.com/kiali/kiali/util"
 )
 
+// AggregateMetrics is the API handler to fetch metrics to be displayed, related to a single aggregate
+func AggregateMetrics(w http.ResponseWriter, r *http.Request) {
+	getAggregateMetrics(w, r, defaultPromClientSupplier)
+}
+
+// getServiceMetrics (mock-friendly version)
+func getAggregateMetrics(w http.ResponseWriter, r *http.Request, promSupplier promClientSupplier) {
+	vars := mux.Vars(r)
+	namespace := vars["namespace"]
+	aggregate := vars["aggregate"]
+	aggregateValue := vars["aggregateValue"]
+
+	prom, namespaceInfo := initClientsForMetrics(w, r, promSupplier, namespace)
+	if prom == nil {
+		// any returned value nil means error & response already written
+		return
+	}
+
+	params := prometheus.IstioMetricsQuery{Namespace: namespace, Aggregate: aggregate, AggregateValue: aggregateValue}
+	err := extractIstioMetricsQueryParams(r, &params, namespaceInfo)
+	if err != nil {
+		RespondWithError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	// Ensure direction=inbound because aggregates always reflect inbound traffic to the target workload.  This ensures
```

Comment:
Isn't it going to produce unexpected/wrong results if params.Direction was outbound? Perhaps return a StatusBadRequest instead?

Reply:
@jotak OK, I can return a StatusBadRequest instead of forcing the Direction.

Reply:
PR Updated.